examples/medplum-chart-demo/src/bots/core/charting-utils.ts

```diff
@@ -0,0 +1,281 @@
+import { getReferenceString } from '@medplum/core';
+import {
+  Bundle,
+  BundleEntry,
+  ClinicalImpression,
+  Coding,
+  Condition,
+  Encounter,
+  Observation,
+  ObservationComponent,
+  Patient,
+  Practitioner,
+  Quantity,
+  Reference,
+  Resource,
+} from '@medplum/fhirtypes';
+import { randomUUID } from 'crypto';
+
+export interface ObservationData {
+  bloodPressure: {
+    systolic?: number;
+    diastolic?: number;
+  };
+  height?: Quantity;
+  weight?: Quantity;
+  bmi?: Quantity;
+  date?: string;
+}
+
+export interface ConditionData {
+  reasonForVisit: Coding;
+  problemList: boolean;
+}
+
+export interface ClinicalImpressionData {
+  visitLength?: number;
+  assessment?: string;
+}
+
+/**
+ * This function takes partial observations with the code and value and fills them out with generic observation data. It then creates a
+ * bundle entry and adds that to an array. These bundle entries can then be used in a batch transaction to create all of the necessary
+ * Observations at once.
+ *
+ * @param observationData - The values to be added to the created Observations
+ * @param encounter - The encounter the observations are derived from
+ * @param user - The user creating the Observations
+ * @param partialObservations - An array of the partial Observations containing the code and value for each given observation
+ * @returns An array of bundle entries which can be added to a batch transaction
+ */
+export function createObservations(
+  observationData: ObservationData,
+  encounter: Encounter,
+  user: Practitioner,
+  partialObservations: Partial<Observation>[]
+): Observation[] {
+  const observations: Observation[] = [];
+  for (const partial of partialObservations) {
+    const code = partial.code;
+    if (!code) {
+      throw new Error('No code provided');
+    }
+    // const request: BundleEntryRequest = {
+    //   method: 'PUT',
+    //   url: code.coding?.[0].code
+    //     ? `Observation?encounter=${getReferenceString(encounter)}&code=${code.coding?.[0].code}`
+    //     : `Observation?encounter=${getReferenceString(encounter)}`,
+    // };
+
+    const observation: Observation = {
+      ...partial,
+      resourceType: 'Observation',
+      status: 'final',
+      code,
+      subject: encounter.subject,
+      encounter: { reference: getReferenceString(encounter) },
+      performer: [{ reference: getReferenceString(user) }],
+      effectiveDateTime: observationData.date,
```

Comment:
since this is the only place we use `observationData`, can you just add this date to the `partial`, and then get rid of the `observationData` input param?